src/main/src/main/java/org/geoserver/catalog/impl/AbstractCatalogFacade.java

```diff
@@ -169,6 +169,16 @@ protected void resolve(LayerGroupInfo layerGroup) {
 
     protected void resolve(StyleInfo style) {
         setId(style);
+
+        // resolve the workspace
+        WorkspaceInfo resolved = ResolvingProxy.resolve(getCatalog(), style.getWorkspace());
+        if (resolved != null) {
+            resolved = unwrap(resolved);
+            style.setWorkspace(resolved);
+        } else {
+            // this means the workspace has not yet been added to the catalog, keep the proxy around
+            System.out.println("this means the workspace has not yet been added to the catalog, keep the proxy around");
```

Comment:
Well, no, we cannot have a System.out here. A log should replace it instead.

Reply:
Indeed. I'll fix the place I copied it from too ( https://github.com/tbarsballe/geoserver/blob/1b4ef13ba465b82b732377ab2661590ab08e3234/src/main/src/main/java/org/geoserver/catalog/impl/AbstractCatalogFacade.java#L180 )